### PR TITLE
[mlir][vector][test] Update tests for vector.xfter_{read|write}

### PR DESCRIPTION
**Pull request by banach-space**
Updates tests in "vector-transfer-permutation-lowering.mlir" to make a
clearer split into cases for :
  * xfer_read vs xfer_write
  * fixed-width vs scalable tests

A new test case is added for fixed-width vectors for `vector.transfer_read`.
This is to complement an existing test for scalable vectors.

This is in preparation for #90835 and also for adding more tests for
scalable vectors.

### CODE REVIEW

File: mlir/test/Dialect/Vector/vector-transfer-permutation-lowering.mlir [hunk 1]
```diff
@@ -29,7 +68,7 @@ func.func @lower_permutation_with_mask_fixed_width(%A : memref<?x?xf32>, %base1
 // CHECK:           %[[TRANSPOSE:.*]] = vector.transpose %[[BCAST]], [0, 2, 1] : vector<8x2x[4]xf32> to vector<8x[4]x2xf32>
 // CHECK:           return %[[TRANSPOSE]] : vector<8x[4]x2xf32>
 // CHECK:         }
-func.func @permutation_with_mask_scalable(%2: memref<?x?xf32>, %dim_1: index, %dim_2: index) -> (vector<8x[4]x2xf32>) {
+func.func @permutation_with_mask_xfer_read_scalable(%2: memref<?x?xf32>, %dim_1: index, %dim_2: index) -> (vector<8x[4]x2xf32>) {
```
Comment (nujaa) [inline mlir/test/Dialect/Vector/vector-transfer-permutation-lowering.mlir:71]: Would adding `permutation_with_mask_xfer_read_fixed_width`  make sense as well ?

**Review by nujaa — APPROVED** [review]
LGTM, simply one debatable comment for consistency.
I'll adapt my PR when this one lands.

**Review by nujaa — APPROVED** [review]

File: mlir/test/Dialect/Vector/vector-transfer-permutation-lowering.mlir [hunk 2]
```diff
@@ -1,23 +1,86 @@
...
+// CHECK-SAME:      %[[IDX_1:.*]]: index,
+// CHECK-SAME:      %[[IDX_2:.*]]: index) -> vector<8x4x2xf32> {
+// CHECK:           %[[C0:.*]] = arith.constant 0 : index
+// CHECK:           %[[PASS_THROUGH:.*]] = arith.constant 0.000000e+00 : f32
+// CHECK:           %[[MASK:.*]] = vector.create_mask %[[IDX_2]], %[[IDX_1]] : vector<2x4xi1>
+// CHECK:           %[[T_READ:.*]] = vector.transfer_read %[[ARG_0]]{{\[}}%[[C0]], %[[C0]]], %[[PASS_THROUGH]], %[[MASK]] {in_bounds = [true, true]} : memref<?x?xf32>, vector<2x4xf32>
+// CHECK:           %[[BCAST:.*]] = vector.broadcast %[[T_READ]] : vector<2x4xf32> to vector<8x2x4xf32>
+// CHECK:           %[[TRANSPOSE:.*]] = vector.transpose %[[BCAST]], [0, 2, 1] : vector<8x2x4xf32> to vector<8x4x2xf32>
+// CHECK:           return %[[TRANSPOSE]] : vector<8x4x2xf32>
+// CHECK:         }
```
Comment (MacDue) [inline mlir/test/Dialect/Vector/vector-transfer-permutation-lowering.mlir:70]: nit: Don't check for `}`
```suggestion
```

File: mlir/test/Dialect/Vector/vector-transfer-permutation-lowering.mlir [hunk 3]
```diff
@@ -1,23 +1,86 @@
...
+// CHECK-SAME:        %[[ARG_0:.*]]: vector<4x[8]xi16>,
+// CHECK-SAME:        %[[ARG_1:.*]]: memref<1x4x?x1xi16>,
+// CHECK-SAME:        %[[MASK:.*]]: vector<4x[8]xi1>) {
+// CHECK:             %[[C0:.*]] = arith.constant 0 : index
+// CHECK:             %[[BCAST_1:.*]] = vector.broadcast %[[ARG_0]] : vector<4x[8]xi16> to vector<1x4x[8]xi16>
+// CHECK:             %[[BCAST_2:.*]] = vector.broadcast %[[MASK]] : vector<4x[8]xi1> to vector<1x4x[8]xi1>
+// CHECK:             %[[TRANSPOSE_1:.*]] =  vector.transpose %[[BCAST_2]], [1, 2, 0] : vector<1x4x[8]xi1> to vector<4x[8]x1xi1>
+// CHECK:             %[[TRANSPOSE_2:.*]] =  vector.transpose %[[BCAST_1]], [1, 2, 0] : vector<1x4x[8]xi16> to vector<4x[8]x1xi16>
+// CHECK:             vector.transfer_write %[[TRANSPOSE_2]], %[[ARG_1]]{{.*}}, %[[TRANSPOSE_1]] {in_bounds = [true, true, true]} : vector<4x[8]x1xi16>, memref<1x4x?x1xi16>
+// CHECK:             return
```
Comment (MacDue) [inline mlir/test/Dialect/Vector/vector-transfer-permutation-lowering.mlir:41]: nit: Don't check for `return`
```suggestion
```

**Review by MacDue — APPROVED** [review]
LG (bar some nits)

P.s. I'd change the tag from `[nfc]` to `[test]` -- especially since you're adding a new test.

**Comment by MacDue** [comment]
(the summary should also be updated to mention the new test case)

**Pull request merged**